pkg/podman/podman.go

```diff
@@ -0,0 +1,188 @@
+package podman
+
+import (
+	"archive/tar"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/containers/buildah/define"
+	"github.com/containers/podman/v4/pkg/bindings"
+	"github.com/containers/podman/v4/pkg/bindings/containers"
+	"github.com/containers/podman/v4/pkg/bindings/images"
+	"github.com/containers/podman/v4/pkg/domain/entities"
+	"github.com/containers/podman/v4/pkg/specgen"
+	"github.com/suse-edge/edge-image-builder/pkg/fileio"
+	"go.uber.org/zap"
+)
+
+const (
+	podmanSock         = "unix:///run/podman/podman.sock"
+	dockerfile         = "Dockerfile"
+	podmanDirName      = "podman"
+	podmanBuildLogFile = "podman-image-build.log"
+)
+
+type Podman struct {
+	context context.Context
```

Comment:
Yea a GitHub issue would need to be raised for this. Essentially we would probably have to do an integration test here as opposed to doing a unit test. Mainly this is true because in order to test the functionality, we need a [podman-system-service](https://docs.podman.io/en/latest/markdown/podman-system-service.1.html) started in order to be able to communicate with podman and execute commands, which will be very hard to mock as a unit test. IMO an integration test would suit this better.